### PR TITLE
TPROD-457 - PHP 8.1 support

### PR DESCRIPTION
**Pull request by roboparker**
#### Description:

Related PRs:
- https://github.com/mautic/core-composer-scaffold/pull/4
- https://github.com/mautic/core-project-message/pull/1
- https://github.com/mautic/mautic/pull/12422
- https://github.com/mautic/mautic/pull/12434
- https://github.com/mautic/mautic/pull/12437
- https://github.com/mautic/mautic/pull/12438
- https://github.com/mautic/mautic/pull/12461
- https://github.com/mautic/mautic/pull/11613

There are a number of issues with upgrading to PHP 8.1. They will be fixed in separate PRS. When all issues are resolved this will be taken out of WIP.

### CODE REVIEW

File: .ddev/config.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: mautic
 type: php
 docroot: ""
-php_version: "8.0"
+php_version: "8.1"
```
Comment (mabumusa1) [inline .ddev/config.yaml:4]: Line #16 should be changed as well to `php8.1-imap`

**Review by mabumusa1 — CHANGES_REQUESTED** [review]
Initial running to the code resulted in these notes, plus the notes that we know from `SendGrid` 

I suggest we remove SendGrid as a package, and its related tests then merge this PR so we can have sometime to run rector rules on the new code.

```

Deprecated: preg_split(): Passing null to parameter #3 ($limit) of type int is deprecated in /var/www/html/vendor/mrclay/minify/min/lib/Minify/CSS/Compressor.php on line 238
PHP Deprecated:  ord(): Passing null to parameter #1 ($character) of type string is deprecated in /var/www/html/vendor/mrclay/minify/min/lib/JSMin.php on line 326

Deprecated: ord(): Passing null to parameter #1 ($character) of type string is deprecated in /var/www/html/vendor/mrclay/minify/min/lib/JSMin.php on line 326
PHP Deprecated:  ord(): Passing null to parameter #1 ($character) of type string is deprecated in /var/www/html/vendor/mrclay/minify/min/lib/JSMin.php on line 326

Deprecated: ord(): Passing null to parameter #1 ($character) of type string is deprecated in /var/www/html/vendor/mrclay/minify/min/lib/JSMin.php on line 326

```

**Comment by escopecz** [comment]
@mabumusa1 I'm not getting these warnings. But I created https://github.com/mautic/mautic/pull/12461 which should fix it. Please review.

**Review by mabumusa1 — APPROVED** [review]
Looks good, works as expected

**Review by escopecz — APPROVED** [review]

**Pull request merged**